src/app/api/webhook/webhook.service.ts

```diff
@@ -56,18 +57,25 @@ class WebhookService extends BaseService {
     return { assigneeId, assigneeType }
   }
 
-  async handleClientCreated(assigneeId: string) {
+  async handleClientCreated(clientId: string, forCompanyId?: string) {
     // First fetch all the current non-complete tasks for client's company, if exists
-    const client = await this.copilot.getClient(assigneeId)
+    const client = await this.copilot.getClient(clientId)
     let company
     try {
       // Accomodate indididual client's company id behavior for copilot
-      company = await this.copilot.getCompany(client.companyId)
+      // Note: When a client is created, only one company can be assigned to it.
+      // This is why we don't need to check for multiple companies here.
+      //
+      // NOTE: forCompanyId is used when a client is created for a company that is not the first company in the client's companyIds array
+      const companyId = forCompanyId || client.companyIds?.[0] || client.companyId
```

Comment:
client.companyIds?.[0] and client.companyId is the same. Furthermore, client.companyId might be deprecated in the future.
```suggestion
      const companyId = forCompanyId || client.companyIds?.[0] 
```

Reply:
@arpandhakal This is for backwards compatibility, which is a good practice to have when making a switch like this.

From left -> right we have the most high priority choice -> low priority one